documentation/openapi-authoring-automated-guidelines.md

```diff
@@ -1024,4 +1029,180 @@ Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rul
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R4000-3" />R4000 DescriptiveDescriptionRequired
+**Output Message**: The value provided for description is not descriptive enough. Accurate and descriptive description is essential for maintaining reference documentation.
+
+**Description**: The value of the 'description' property must be descriptive.
+
+**Why the rule is important**: This will ensure that the service is well documented.
+
+**How to fix the violation**: For each 'description' property, provide a detailed description value.
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R4000-4" />R4000 DescriptionAndTitleMissing
+**Output Message**: {0} lacks 'description' and 'title' property. Consider adding a 'description'/'title' element. Accurate description/title is essential for maintaining reference documentation.
+
+**Description**: The definition must have atleast one of the property - description/title.
+
+**Why the rule is important**: This will ensure that the definition is well documented.
+
+**How to fix the violation**: For each definition, provide atleast one of the property - description/title.
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R4000-5" />R4000 OperationDescriptionOrSummaryRequired
+**Output Message**: {0} lacks 'description' and 'summary' property. Consider adding a 'description'/'summary' element. Accurate description/summary is essential for maintaining reference documentation.
+
+**Description**: Every operation must have a 'description'/'summary' property.
+
+**Why the rule is important**: This will ensure that the operation is well documented.
+
+**How to fix the violation**: For each operation, provide atleast one of the property - description/summary.
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R3013" />R3013 DeleteMustNotHaveRequestBody
+**Output Message**: 'Delete' operation '{0}' must not have a request body.
+
+**Description**: The request body of a delete operation must be empty.
+
+**Why the rule is important**: This will ensure that the delete operation aligns with the RPC guidelines.
```

Comment:
ARM guidelines, and you can link to the guidelines (you can pick up the generic link from another rule)